src/interfaces/dispenser.cpp

```diff
@@ -79,7 +79,10 @@ namespace
 
             mdhandle_ptr md_ptr{ mdhandle };
 
-            dncp::com_ptr<ControllingIUnknown> obj { new (std::nothrow) ControllingIUnknown() };
+            dncp::com_ptr<ControllingIUnknown> obj;
+            // ControllingIUnknown's constructor assumes that we start with 1 ref. dncp::com_ptr always calls AddRef when created with a pointer,
+            // but not when assigned the underlying pointer in this manner.
+            obj.p = new (std::nothrow) ControllingIUnknown();
```

Comment:
This is precisely what the `com_ptr::Attach()` API is for. I would remove the comment, the `Attach()` semantics are well know for `CComPtr<T>`.
```suggestion
            obj.Attach(new (std::nothrow) ControllingIUnknown());
```